src/test/java/mondrian/test/DialectTest.java

```diff
@@ -1545,6 +1546,32 @@ public void testMonetBooleanColumn() throws SQLException {
         assertEquals(SqlStatement.Type.OBJECT, type);
     }
 
+    public void testsqlTypeToDatatype() throws SQLException {
+        Dialect dialect = new JdbcDialectImpl();
+
+        assertEquals( Datatype.Integer, dialect.sqlTypeToDatatype("BIT", Types.BIT) );
+
+        assertEquals( Datatype.Date, dialect.sqlTypeToDatatype("DATE", Types.DATE) );
+
+        assertEquals( Datatype.Boolean, dialect.sqlTypeToDatatype("BOOLEAN", Types.BOOLEAN) );
+
+        assertEquals( Datatype.Integer, dialect.sqlTypeToDatatype("TINYINT", Types.TINYINT) );
+        assertEquals( Datatype.Integer, dialect.sqlTypeToDatatype("SMALLINT", Types.SMALLINT) );
+        assertEquals( Datatype.Integer, dialect.sqlTypeToDatatype("INT", Types.INTEGER) );
+        assertEquals( Datatype.Integer, dialect.sqlTypeToDatatype("BIGINT", Types.BIGINT) );
+
+        assertEquals( Datatype.Numeric, dialect.sqlTypeToDatatype("DOUBLE", Types.DOUBLE) );
+        assertEquals( Datatype.Numeric, dialect.sqlTypeToDatatype("FLOAT", Types.FLOAT) );
+        assertEquals( Datatype.Numeric, dialect.sqlTypeToDatatype("DECIMAL", Types.DECIMAL) );
+        assertEquals( Datatype.Numeric, dialect.sqlTypeToDatatype("FLOAT", Types.REAL) );
+
```

Comment:
It could be since the type name really isn't used in the function at all.  However, Mysql returns the type name as "FLOAT" for a jdbc Types.REAL.  I simply imitated what Mysql would do.